### PR TITLE
[FLOC-3493] Add ATTACHED_ELSEWHERE state.

### PR DESCRIPTION
**Pull request by tomprince**
#2201 is an example of where we will need access to this state.

### CODE REVIEW

File: flocker/node/agents/test/test_blockdevice.py [hunk 1]
```diff
@@ -896,7 +896,14 @@ def test_only_remote_device(self):
         )
         assert_discovered_state(
             self, self.deployer,
-            expected_discovered_datasets=[],
+            expected_discovered_datasets=[
```
Comment (itamarst) [inline flocker/node/agents/test/test_blockdevice.py]: Need to change the test's docstring too.

**Comment by itamarst** [comment]
Seems like you'd want these to go in NonManifestDatasets for backwards compat with the current state of the logic (pre-start of refactor)? Even if deployer is going to stop looking at that, the volume catalog is still going to rely on that for a while until we have better model in control service.

**Comment by itamarst** [comment]
Or, hm. We're going to end up with duelling updates to non-manifest datasets which is going to result in confusing user experience, until the control service data model is updated. How did we handle this before? Or did we have the same issue?

**Comment by itamarst** [comment]
Apparently current state of things is non-manifest changing randomly and some datasets disappearing. Luckily datasets usually don't end up in this edge case since REST API just has destination...

So, merge when green (one minor comment above) and if there's no extra exceptions in acceptance test logs.
